qemu/tests/rng_host_guest_read.py

```diff
@@ -0,0 +1,73 @@
+import re
+import logging
+
+from virttest import error_context
+from virttest import utils_test
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu virtio-rng device test:
+    1) boot guest with virtio-rng device
+    2) host read random numbers in the background
+    3) guest read random data at the same time during step2
+    4) clean host read process after test
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def host_read_start(host_read_cmd):
+        """
+        Read random numbers on the host side
+        :param host_read_cmd: host read random numbers command line
+        :return: reture host_read_process
+        """
+        host_read_process = process.SubProcess(host_read_cmd, shell=True)
+        host_read_process.start()
+        return host_read_process
+
+    def is_host_read_running(target_process):
+        """
+        Check host reading test is alive.
+
+        :param target_process: targeted host reading process
+        :return: return True if host is reading random numbers;
+                 else return False
+        """
+        error_context.context("Check whether host is reading is alive",
+                              logging.info)
+        host_read_check_cmd = params.get("host_read_check_cmd")
+        output = process.system_output(host_read_check_cmd, shell=True)
+        result = re.findall(target_process, output, re.M | re.I)
+        return bool(result)
+
+    def host_read_stop(host_read_process):
+        """
+        Clean host read random numbers process
+        :param host_read_process: process of the host reading
+        """
+        if is_host_read_running(host_read_cmd):
+            host_read_process.stop()
+
+    host_read_cmd = params.get("host_read_cmd")
+    rng_test = params.get("rng_test")
+    vm = env.get_vm(params["main_vm"])
+    vm.wait_for_login()
+
+    error_context.context("Host read random numbers in the background",
+                          logging.info)
+    host_read_process = host_read_start(host_read_cmd)
+
+    if is_host_read_running(host_read_cmd):
+        error_context.context("Guest begin to read random numbers",
+                              logging.info)
+        utils_test.run_virt_sub_test(test, params, env, rng_test)
```

Comment:
Is it better to change parameter rng_test's name to guest related for this case?
such as guest_read_test or guest_rng_test

Reply:
Modified.
Thanks